### DISCUSSION TITLE
Athens 12.2

### DISCUSSION THREAD

**Discussion by acgubamg** [project-jedi/jedi#20] [Ideas]
What do I need to change to install in the Athens 12.2 IDE?

**Comment by obones** [comment]
AFAIK, nothing, it should be recognized as Delphi 12.

**Comment by acgubamg** [comment]
JCL installed ok.
jvcl: ---------------------------
dcc32ex.exe
---------------------------
No Delphi/BCB/BDS/RAD-Studio versions was found that has the required

dependencies installed. Please install the dependencies first.
---------------------------
OK   
---------------------------

**Comment by acgubamg** [comment]
I Clear jcl install, its solved

**Comment by oliwe** [comment]
JCL and JVCL can not be uses in C++Builder with then new bcc64x compiler. Is support planned?

**Comment by obones** [comment]
Hello
From what I read, up until the very latest version of this new compiler, packages were not supported and thus the JCL/JVCL would not be usable anyway as they are provided as Delphi based packages imported by the bcc compiler.
If this now works, then someone should be able to compile those packages with dcc32 and then use them within a project using bcc64x via the generated HPP files.
But as I don't have such an installation, I cannot confirm if it's viable, nor if it would match your expectations. In particular, what do you mean by "can not be used"? Does this imply automatic installation? manual installation? something else?

**Reply by oliwe** [reply]
Hello - I have installed 12.2.
Thank you for the answer!

To always have the latest versions of JCL and JVCL, I have downloaded the sources from GitHub. Installing both in the IDE was problem-free. I am a C++ programmer. With the old toolchain (bcc64) you can use JCL and JVCL in the IDE - with the new one (bcc64x) the JVCL components are grayed out and cannot be used. Is there a solution?

**Reply by oliwe** [reply]
"If this now works, then someone should be able to compile those packages with dcc32 and then use them within a project using bcc64x via the generated HPP files." -> Can you give more details - I never did use dcc32 directly.

**Reply by obones** [reply]
Well, if you see them in the IDE, then they were already compiled by the installer and so there should not be any need to do anything manually.
Figuring out why they are greyed out is the next step and, from memory, this comes from a missing element in the dproj for any given package. They are generated from a template, so any pull request will have to include the change in both the packages and the template.
Maybe it's an element in the Platforms node, not sure.

**Reply by hansw91** [reply]
Hello,
according to [https://docwiki.embarcadero.com/RADStudio/Athens/en/Build_Delphi_Packages_for_C%2B%2B](https://docwiki.embarcadero.com/RADStudio/Athens/en/Build_Delphi_Packages_for_C%2B%2B) a new target platform has to be added to all Delphi projects. Then building the packages with the "-jf:coffi" option will generate COFF-format C++ files for the Win64x platform.
I do not have a Delphi license. But from looking at other packages it seems the <TargetedPlatforms> section must be set to 1048579 (instead of 3), furthermore every <PropertyGroup> with "Win64" has to occur also as "Win64x", and in <Platforms> "Win64x" must be set to "True".
And the Win64x platform has to use output folders that are different from Win64.

**Comment by npnneykov** [comment]
Hello,
I had no problems with RAD Delphi Pro12.2. But with the latest version, I had problems installing on RAD 12.3 Delphi Professional. A problem with the double packages option. I removed this option. The next problem - the HPP files - it asks for a bcc compiler. When I removed these options - it now installs normally. But ... it doesn't visible with the new 64 bit IDE.
Will there be a solution soon to appear in the new IDE?
Best regards

**Reply by obones** [reply]
Please create a separate thread in  [JCL discussions](https://github.com/project-jedi/jcl/discussions), it would make things clearer.
